### ISSUE TITLE
LabeledExpression false positive with nested same-named implicit scopes

### ISSUE THREAD

**Issue by TWiStErRob** [detekt/detekt#5131]
## Steps to Reproduce

Very contrived minimal repros:
```kotlin
"outer".apply s@{
	"inner".apply {
		println(this@s)
	}
}
```
Would also apply to `run`, `with`, @DslMarker stuff etc.

```kotlin
"outer".forEach s@{
	"inner".forEach {
		if (...) return@s // === break@s but for stdlib inline funs.
	}
}
```
Would also apply to any other side-effect funs e.g. forEachIndexed and maybe onEach and any other user defined ones.

## Expected Behavior
No report.

## Observed Behavior
> LabeledExpression - [Expression with labels increase complexity and affect maintainability.]

## Context
This is the only way to reference the outer object bar a full refactor of the code. In this case removing the label as recommended will result in compilation error. Real example: creating objects with scoping functions and backreferences to higher level objects
```kotlin
View().apply view@{
	graphId = result.graphId
	date = fixtTime.toInstant()
	atCinema = fixtCinema.apply { views = mutableSetOf(this@view) }
	watchedFilm = fixtFilm.apply { views = mutableSetOf(this@view) }
	userRef = fixtUser.apply { views = mutableSetOf(this@view) }
}
```

## Your Environment

* Version of detekt used: 1.21.0
* Version of Gradle used (if applicable): 7.5
* Gradle scan link (add `--scan` option when running the gradle task): N/A
* Operating System and version: N/A
* Link to your project (if it's a public repository): https://github.com/TWiStErRob/net.twisterrob.cinema/blob/1a45537b0b06b69816d4ede8bd7f2e675fb4efc9/Heroku/backend/database/src/test/kotlin/net/twisterrob/cinema/database/model/ViewIntgTest.kt#L33-L50

**Comment by TWiStErRob** [comment]
Doesn't even need to be same-named: it's enough to combine a fun receiver with a coroutine.
```
private fun OutgoingContent.asString(): String? =
	runBlocking {
		val channel = ByteChannel(true)
		(this@asString as OutgoingContent.WriteChannelContent).writeTo(channel)
		channel.tryReadText(Charsets.UTF_8)
	}
```

**Comment by BraisGabin** [comment]
I understand the issue but the rule flags *all*  the labeled expressions with the idea that "a solution without labels will be less complex". I think that's in general true but for sure it's not always true.

I think that the only way here is to suppress it in the cases where any refactor will make the core less readable or disable the rule if you don't agree with it (I don't have it enabled in my projects).

**Comment by TWiStErRob** [comment]
Same as the other issue, I misunderstood the intent because of the "exception about outer class labels".

**Comment by TWiStErRob** [comment]
@BraisGabin bit of background: I use `allRules = true` and then [ignore](https://github.com/TWiStErRob/net.twisterrob.cinema/blob/master/Heroku/config/detekt/detekt.yml) whatever I don't agree with. It helps me discover new rules, think about my style and _hopefully_ help improve Detekt by flagging issues.

**Comment by BraisGabin** [comment]
It helps a lot 👍 please keep bringing up these type of issues :).
